api/wallet.go

```diff
@@ -371,13 +371,20 @@ func (srv *Server) walletTransactionHandler(w http.ResponseWriter, req *http.Req
 
 // walletTransactionsHandler handles API calls to /wallet/transactions.
 func (srv *Server) walletTransactionsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	// If 'startheight' or 'endheight' query parameters don't exist,
+	// return a descriptive error message along with a HTTP 400 Bad Request response code.
+	startheightStr, endheightStr := req.FormValue("startheight"), req.FormValue("endheight")
+	if startheightStr == "" || endheightStr == "" {
+		writeError(w, "startheight and endheight must be provided to a /wallet/transactions call.", http.StatusBadRequest)
+		return
+	}
 	// Get the start and end blocks.
-	start, err := strconv.Atoi(req.FormValue("startheight"))
+	start, err := strconv.Atoi(startheightStr)
 	if err != nil {
 		writeError(w, "error after call to /wallet/transactions: "+err.Error(), http.StatusBadRequest)
```

Comment:
I think the error string can still be useful, as `Atoi` can return one of a few different errors. The "error after call to /wallet/transactions:" string seems unnecessary though as callers should know what endpoint they're requesting.